system/Images/Handlers/BaseHandler.php

```diff
@@ -723,7 +727,7 @@ public function __call(string $name, array $args = [])
      * This function lets us re-proportion the width/height
      * if users choose to maintain the aspect ratio when resizing.
      */
-    protected function reproportion()
+    protected function reproportion(): void
```

Comment:
Sorry, can you make this as PHPDoc instead? I mistakenly written yesterday in your earlier PR that it is protected so we can add the `: void` directly. I meant it is `final`. With your change it will violate LSP for those people extending the methods.

Please revise the same with the other protected methods.

Reply:
Should the image exception return type also be written in the comments?

Reply:
@MGatner @paulbalandan 
> Framework exceptions are explicitly for framework code - so even though they aren't marked @internal they still should not be extended

It is not documented, and I did not know that.
In fact, we could extend these exceptions.

So I oppose this breaking change. It is better to add `@internal` or `final` in `4.4` branch, and add return type.